resources/prosody-plugins/mod_token_verification.lua

```diff
@@ -50,25 +52,17 @@ local function verify_user(session, stanza)
 		return nil;
 	end
 
-	local room = string.match(stanza.attr.to, "^(%w+)@");
-	log("debug", "Will verify token for user: %s, room: %s ", user_jid, room);
-	if room == nil then
-		log("error",
-			"Unable to get name of the MUC room ? to: %s", stanza.attr.to);
-		return nil;
-	end
-
 	local token = session.auth_token;
 	local auth_room = session.jitsi_meet_room;
-	if disableRoomNameConstraints ~= true and room ~= string.lower(auth_room) then
+	local auth_domain = session.jitsi_meet_domain;
+	if disableRoomNameConstraints ~= true and token_util.verify_room_and_domain(stanza.attr.to, auth_room, auth_domain) ~= true then
```

Comment:
I think that token_util may not be getting right config values for "mod_token_verification" which is supposed to load them for the parent host. Or at least it's hard to figure out just by looking at the code.